### DISCUSSION TITLE
Cannot generate installation access token even though installation id is correct

### DISCUSSION THREAD

**Discussion by hamelsmu** [community/community#22978] [New to GitHub]
<p>I’m confused as to why I cannot generate an installation access token.  See the below two commands:</p>
<pre><code>&gt; curl -i -H "Authorization: Bearer &lt;JWT&gt; -H "Accept: application/vnd.github.machine-man-preview+json" https://api.github.com/app/installations/754036

{
"id": 754036,
"account": {
"login": "hamelsmu",
"id": 1483922,
"node_id": "MDQ6VXNlcjE0ODM5MjI=",
"avatar_url": "https://avatars2.githubusercontent.com/u/1483922?v=4",
"gravatar_id": "",
"url": "https://api.github.com/users/hamelsmu",
"html_url": "https://github.com/hamelsmu",
"followers_url": "https://api.github.com/users/hamelsmu/followers",
"following_url": "https://api.github.com/users/hamelsmu/following{/other_user}",
"gists_url": "https://api.github.com/users/hamelsmu/gists{/gist_id}",
"starred_url": "https://api.github.com/users/hamelsmu/starred{/owner}{/repo}",
"subscriptions_url": "https://api.github.com/users/hamelsmu/subscriptions",
"organizations_url": "https://api.github.com/users/hamelsmu/orgs",
"repos_url": "https://api.github.com/users/hamelsmu/repos",
"events_url": "https://api.github.com/users/hamelsmu/events{/privacy}",
"received_events_url": "https://api.github.com/users/hamelsmu/received_events",
"type": "User",
"site_admin": true
},
"repository_selection": "selected",
"access_tokens_url": "https://api.github.com/installations/754036/access_tokens",
"repositories_url": "https://api.github.com/installation/repositories",
"html_url": "https://github.com/settings/installations/754036",
"app_id": 27079,
"target_id": 1483922,
"target_type": "User",
"permissions": {
"issues": "write",
"pull_requests": "write",
"app_config": "read",
"checks": "read",
"contents": "read",
"metadata": "read"
},
"events": [
"issues",
"project",
"project_card",
"project_column",
"pull_request",
"pull_request_review",
"pull_request_review_comment"
],
"created_at": "2019-03-16T04:57:54.000Z",
"updated_at": "2019-03-16T17:03:35.000Z",
"single_file_name": null
}
</code></pre>
<p>However when I add _ <strong>access_tokens</strong> _ to the end of the url it returns the following:</p>
<pre><code>&gt; curl -i -H "Authorization: Bearer &lt;JWT&gt;" -H "Accept: application/vnd.github.machine-man-preview+json" https://api.github.com/app/installations/754036/access_tokens

HTTP/1.1 404 Not Found
Date: Sun, 17 Mar 2019 05:20:25 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 87
Server: GitHub.com
Status: 404 Not Found
X-GitHub-Media-Type: github.machine-man-preview; format=json
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
X-GitHub-Request-Id: C82A:9BB9:1AA2AFF:207EC5B:5C8DD919

{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3"
}
</code></pre>
<p>Does anyone have any idea about this?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Thank you for being here! Could you please send us the full output of a <code>curl -v</code> request that demonstrates the problem?</p>
<p><a href="http://curl.haxx.se/" rel="nofollow noopener">http://curl.haxx.se/</a></p>
<p>That should help us investigate the issue. Also, please make sure you mask any sensitive information like OAuth tokens and Authorization headers in the output of the curl command. Please send the output to <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a>. And include a link to this thread to give support some background on the ticket.</p>
<p>I hope this helps!</p>
<p>Best,<br>
Andrea</p>

**Comment by hamelsmu** [comment, marked as answer]
<p>This issue was solved because I made a mistake in the request, </p>
<p>the <code>/access_tokens</code> endpoint should be accessed via a <code>POST</code> request, and I was using <code>GET</code>.  Per the documentation, this is done with </p>
<pre><code>curl -i -X POST \
-H "Authorization: Bearer YOUR_JWT" \
-H "Accept: application/vnd.github.machine-man-preview+json" \
https://api.github.com/app/installations/:installation_id/access_tokens
</code></pre>
<p>I just didn’t see the <code>POST</code> part when I was reading it.  Thanks</p>

**Comment by AndreaGriffiths11** [comment]
<p>Thanks so much, <a class="mention" href="/hamelsmu">@hamelsmu</a> for coming back to share what worked!</p>
